### ISSUE TITLE
Double object construction in Welsh

### ISSUE THREAD

**Issue by ftyers** [UniversalDependencies/UD_Welsh-CCG#3]
In [this issue](https://github.com/UniversalDependencies/UD_Welsh-CCG/issues/1#issuecomment-390899220) @jheinecke brings up the question of what should we do with the double object construction in Welsh.

Consider a sentence like:

![40349483-0b4edd20-5da7-11e8-9efd-6dfca625f053](https://user-images.githubusercontent.com/449545/40360186-840dae80-5dc5-11e8-9e96-ef4101865e27.png)

As far as I remember, one of those object `PRON` can be replaced by a full NP, and that is the one I would say should get the `obj` label, while the second one could get an `expl` label.

**Comment by jheinecke** [comment]
In case of verbnouns, the first or the second `PRON` can be absent. In this case the case is clear.
Same syntactical structure, but for nouns which are no verbnouns
- _fy nhŷ i_ (my house)
- _dy tŷ ti_ (your (sing) house)

so both pronouns are simple possessives. So possibly we may drop the `obj `relation and use `nmod:poss` in all these cases ?

**Comment by donnekgit** [comment]
There are three forms of possessive in Welsh, applying to both nouns and infinitives/verbal nouns:
(a) preceding: *dy lyfr* (your book)
(b) following: *llyfr ti* 
(c) sandwich: *dy lyfr di* (I think with 2PS it would be more common to have the mutated form (*di*) rather than the unmutated form as in your example above)

The "standard" one is the sandwich, with the fallback being a preceding possessive, so the template might be: POSS ... [POSS].  A following possessive was/is generally considered "sub-standard", but seems to be gaining ground, and not only among the young, especially for 3PS - see Davies (2016) - possibly because it mirrors the typical genitive construction in Welsh (eg *llyfr Bob*, Bob's book).

For the sandwich, I think both should be marked with the same tag, given that either side can be omitted (though in practice it's the right side that is (currently) most often omitted).

> dy tŷ ti (your (sing) house)

Probably better to write *dy dŷ di*, but for stylistic reasons you're more likely to hear just *dy dŷ*.

Davies, P. (2016). Age variation and language change in Welsh: Auxiliary deletion and possessive
constructions. In M. Durham & J. Morris (Eds.), Sociolinguistics in Wales. (31–59). London:
Palgrave Macmillan. doi: 10.1057/978-1-137-52897-1_2

**Comment by ftyers** [comment]
@jheinecke,

> so both pronouns are simple possessives. So possibly we may drop the objrelation and use nmod:poss in all these cases ?

Well, it depends on if you consider the _garu_ there as a clause/predicate or not. If it is a clause then it should get `[nc]subj`, `obj` and friends. If it is a nominal then it could get `nmod:poss`. 

The default way of doing subordination in the Turkic languages is nominalisation, for example (sentence is in Turkish and slightly bookish):
```
senin   şarkı   söyleyerek   bahçede      oynadığını                gördüm
you-GEN song    say-Vadv     garden-LOC   play-Vnoun-POSS.2SG-ACC   see-PAST-1SG 

I saw you play in the garden while singing (lit. "I saw your playing in the garden while singing")
```

Here in the Turkic languages, _oynadığını_ is definitely a nominal form, the subject is in the genitive and the verb shows possessive agreement with the subject. But it is also definitely verbal because it allows for clausal modification by verbal adverbs, _şarkı söyleyerek_ "while singing" and allows direct modification by other locative case forms _bahçede_, Turkic NPs don't allow direct modification by the locative case, you need the _-ki_ morpheme to make them attributive.

I'm not sure what kind of diagnostics can be used in Welsh, it could be that it isn't possible to distinguish clauses/predicates from nominals as well as Turkic, but it would be good to reflect on possible ways to distinguish them.

**Comment by jheinecke** [comment]
I am not strong enough in Turkish to judge this nice example. But in Welsh you can adverbials to verbnouns (and not to nouns)
```
Mae o  'n        gweithio 'n       dda/llawer/*goch
is  he yn_aspect work-VN   yn_pred well/lots/*red

he is working well/a lot/*red
```
But compare with this
```
Mae 'r  car yn      dda/*llawer/goch
is  DEF car yn_pred good/*a lot/red

the car is good/*a lot/red
```

It looks similar, but in the second case it is a noun (and in the translation _dda_ is translated with the English adjective _good_ instead of the adverb _well_). The difference is in the choice of yn-phrases.
I have not yet had the time to think in detail of all of this. In any case it is not the English translation which counts :-)

**Comment by ftyers** [comment]
@jheinecke yes, we certainly shouldn't be basing anything on the English translation. What kind of modification strategies are allowable for each type is however a good diagnostic. 

The purpose of the Turkish example was to show that verbal nouns in the Turkic languages can take nominal strategies for modification (genitive + possession) but the thing that distinguishes them from non-clausal nouns is that they are also open to clausal modification strategies (verbal adverbs and locative PPs), which non-clausal nouns are not. 

Distinguishing clauses from non-clauses is central to the UD relation taxonomy so it's something that needs to be determined fairly early on.
